generators/bootstrap-application-base/generator.mts

```diff
@@ -0,0 +1,191 @@
+/**
+ * Copyright 2013-2022 the original author or authors from the JHipster project.
+ *
+ * This file is part of the JHipster project, see https://www.jhipster.tech/
+ * for more information.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import _ from 'lodash';
+
+import BaseApplicationGenerator from '../base-application/index.mjs';
+import entityUtils from '../../utils/entity.js';
+import fieldUtils from '../../utils/field.js';
+import relationshipUtils from '../../utils/relationship.js';
+import utils from '../../utils/index.js';
+import userUtils from '../../utils/user.js';
+import constants from '../generator-constants.js';
+import type { CommonClientServerApplication } from './types.js';
+
+const { prepareEntityForTemplates } = entityUtils;
+const { prepareFieldForTemplates } = fieldUtils;
+const { prepareRelationshipForTemplates } = relationshipUtils;
+const { stringify } = utils;
+const { createUserEntity } = userUtils;
+const { upperFirst } = _;
+const { NODE_VERSION } = constants;
+
+/**
+ * @class
+ * @extends { BaseApplicationGenerator<CommonClientServerApplication> }
+ */
+export default class BootStrapApplicationBase extends BaseApplicationGenerator<CommonClientServerApplication> {
+  constructor(args: any, options: any, features: any) {
+    super(args, options, { unique: 'namespace', ...features });
+
+    if (this.options.help) return;
+
+    this.loadStoredAppOptions();
+  }
+
+  get loading() {
+    return this.asLoadingTaskGroup({
+      loadApplication({ application }) {
+        this.loadAppConfig(undefined, application);
+        this.loadTranslationConfig(undefined, application);
+      },
+    });
+  }
+
+  get [BaseApplicationGenerator.LOADING]() {
+    return this.loading;
+  }
+
+  get preparing() {
+    return this.asPreparingTaskGroup({
+      prepareApplication({ application }) {
+        this.loadDerivedAppConfig(application);
+
+        application.nodePackageManager = 'npm';
+        application.nodeDestinationVersion = NODE_VERSION;
+
+        // TODO v8 drop the following variables
+        const anyApplication = application as any;
+
+        anyApplication.clientPackageManager = application.nodePackageManager;
+        anyApplication.protractorTests = false;
```

Comment:
Protractor tests should be removed also?

Reply:
Leave to avoid templates breakage
We have an open issue for this one, https://github.com/jhipster/generator-jhipster/issues/19484.